mod/blog/actions/blog/auto_save_revision.php

```diff
@@ -40,7 +40,7 @@
 
 		// mark this as a brand new post so we can work out the
 		// river / revision logic in the real save action.
-		$blog->new_post = true;
+		$blog->new_post = 1;
```

Comment:
why?

Reply:
Metadata can't hold bools, but anyway this should be `"1"` for perfect BC. That's what gets stored.

Reply:
i do not see the need for this change... metadata will cast it for you to a `1`.. this is not making things more/ or less BC... we are storing booleans throughout the whole system... why specifically fix this here?

Reply:
Are typed `@property` declarations for metadata should either be `int` or `string`. Otherwise we are fooling ourselves and new devs who look at the code and assume what was set is what is returned.

Reply:
As a compromise I could leave the `@property` as string (what gets returned) but leave the set code as is.

Reply:
fine...

Reply:
if you have a problem with setting `true` and getting `1` we can discuss this in a new ticket, because i rather see that boolean becomes a valid type for metadata, rather than changing this kind of stuff randomly... changing this right would be unrelated to the intentions of this PR

Reply:
It's not much trouble adding boolean support into QueryBuilder

Reply:
that is good to hear, however i expect issue with migrating the data as @mrclay pointed out